### PR TITLE
feat: ✨ add simulated register data to `targets` pipeline

### LINKED ISSUE #134: The "honuge" variable in the fake health service data should be formatted as a 4-digit string
Right now it looks like any initial zeroes are removed, so I assume it's a int or dbl (see data-sources.html)

### LINKED ISSUE #166: Add register_data to `targets` pipeline

### PR DESCRIPTION
**Pull request by signekb**
This PR adds the recreation of simulated register data to our `targets` pipeline and runs the updated pipeline.

For this, I've: 
- Added a function to read to simulations definitions: `read_simulation_definitions_data()`, in the same style as the `read_` functions for `variable_description` and `algorithm`
- Rewritten the creation of simulated data to a function: `create_simulated_data`
- Added this to the pipeline in `_targets.R`
- Rerun the pipeline

Closes #166 
Closes #134 (bc now the updates from #165 will be run)

### CODE REVIEW

File: data-raw/simulate-data.R [hunk 1]
```diff
@@ -411,27 +411,39 @@ simulate_data <- function(data, n) {
...
-  map(add_fake_drug_name) |>
-  map(insert_false_drug_names) |>
-  map(insert_false_metformin) |>
-  map(insert_analysiscode) |>
-  # add the register abbreviations as a name to the list
-  set_names(
-    map_chr(
-      simulation_definitions_list,
-      \(data) unique(data$register_abbrev)
+set.seed(123)
```
Comment (signekb) [inline data-raw/simulate-data.R:425]: create_simulated_data

File: _targets.R [hunk 2]
```diff
@@ -53,10 +54,37 @@ list(
...
+  ),
+  tar_target(
+    name = simulation_definitions,
+    command = read_simulation_definitions_data(simulation_definitions_csv)
+  ),
+  tar_target(
+    name = register_data,
+    command = {
+      create_simulated_data(simulation_definitions)
+    },
```
Comment (lwjohnst86) [inline _targets.R:70]:
```suggestion
    command = create_simulated_data(simulation_definitions),
```

File: data-raw/simulate-data.R (hunk 1 above)
Comment (lwjohnst86) [reply data-raw/simulate-data.R:425]: No, not if it is outside the function that needs the set seed.

File: data-raw/simulate-data.R [hunk 3]
```diff
@@ -411,27 +411,39 @@ simulate_data <- function(data, n) {
...
+    map(insert_false_drug_names) |>
+    map(insert_false_metformin) |>
+    map(insert_analysiscode) |>
+    # add the register abbreviations as a name to the list
+    set_names(
+      map_chr(
+        simulation_definitions_list,
+        \(data) unique(data$register_abbrev)
+      )
     )
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:448]: Instead of two functions, this could be one (`create_simulated_data()`).

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Just some small suggestions :relaxed:

File: data-raw/simulate-data.R (hunk 1 above)
Comment (signekb) [reply data-raw/simulate-data.R:425]: But isn't it bad practice to set a seed inside a function? Maybe not?

File: data-raw/simulate-data.R (hunk 3 above)
Comment (signekb) [reply data-raw/simulate-data.R:448]: You mean having the functionality of `read_simulated_data()` inside `create_simulated_data()`? 
I did it in two separate functions to mirror the functions for reading the variable descriptions and algorithm.

Comment (signekb) [reply data-raw/simulate-data.R:448]: I'll merge the functions into one 👍

File: data-raw/simulate-data.R (hunk 1 above)
Comment (signekb) [reply data-raw/simulate-data.R]: (I've moved it inside the function now)

Comment (lwjohnst86) [reply data-raw/simulate-data.R]: Or set the seed in the targets pipeline: https://books.ropensci.org/targets/random.html#reproducibility

**Review by lwjohnst86 — APPROVED** [review]
:rocket: :guitar:

File: data-raw/simulate-data.R (hunk 1 above)
Comment (signekb) [reply data-raw/simulate-data.R]: Done!

**Review by Aastedet — APPROVED** [review]
Awesome. I think this also closes #170 , but let's see once it's merged to main.

**Pull request merged**